processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java

```diff
@@ -214,9 +214,8 @@ public void write(T objectToWrite) throws IOException
     }
 
     ++numWritten;
-    // for compatibility with the format (see GenericIndexed javadoc for description of the format), but this field is
-    // unused.
-    valuesOut.writeInt(0);
+
```

Comment:
Please keep the comment about compatibility and that currently this unused field is used as opportunistically to store nullness marker, but in a better format this info should take 1 bit.

Reply:
added back.